src/modules/always_on_top/dllmain.cpp

```diff
@@ -0,0 +1,265 @@
+  HWND                     currentlyOnTop{ nullptr };
+  AllwaysOnTopSettings     settings;
+  std::wstring             itemName{ KMenuItemName };
+  PowertoySystemMenuIface* systemMenuHelper{ nullptr };
+
+  void ProcessCommand(HWND window);
+  bool SetWindowOnTop(HWND window);
+  void ResetCurrentOnTop();
+
+  void LoadSettings(PCWSTR config, bool fromFile);
+  void SaveSettings();
+
+  std::vector<PowertoySystemMenuIface::ItemInfo> CustomItems();
+
+  LRESULT WndProc(HWND, UINT, WPARAM, LPARAM) noexcept;
+
+protected:
+
+  static LRESULT CALLBACK WndProc_Helper(HWND, UINT, WPARAM, LPARAM) noexcept;
+
+public:
+
+  AlwaysOnTop()
+  {
+    LoadSettings(MODULE_NAME, true);
+  }
+
+  virtual void destroy() override
+  {
+    ResetCurrentOnTop();
+    if (hotKeyHandleWindow) {
+      DestroyWindow(hotKeyHandleWindow);
+      hotKeyHandleWindow = nullptr;
+    }
+    delete this;
+  }
+
+  virtual const wchar_t* get_name() override
+  {
+    return MODULE_NAME;
+  }
+
+  virtual const wchar_t** get_events() override
+  {
+    static const wchar_t* events[] = { nullptr };
+    return events;
+  }
+
+  virtual bool get_config(wchar_t* buffer, int* buffer_size) override
+  {
+    HINSTANCE hinstance = reinterpret_cast<HINSTANCE>(&__ImageBase);
+
+    PowerToysSettings::Settings common(hinstance, get_name());
+    common.add_hotkey(HOTKEY_NAME, IDS_SETTING_ALWAYS_ON_TOP_HOTKEY, settings.editorHotkey);
+
+    return common.serialize_to_buffer(buffer, buffer_size);
+  }
+
+  virtual void set_config(const wchar_t* config) override
+  {
+    LoadSettings(config, false);
+    SaveSettings();
+
+    PowerToysSettings::PowerToyValues values =
+      PowerToysSettings::PowerToyValues::from_json_string(config);
+    if (values.is_object_value(HOTKEY_NAME)) {
+      settings.editorHotkey = PowerToysSettings::HotkeyObject::from_json(values.get_json(HOTKEY_NAME));
+    }
+
+    UnregisterHotKey(hotKeyHandleWindow, 1);
+    RegisterHotKey(hotKeyHandleWindow, 1, settings.editorHotkey.get_modifiers(), settings.editorHotkey.get_code());
+
+    // Update hotkey.
+    systemMenuHelper->SetConfiguration(this, CustomItems());
+  }
+
+  virtual void call_custom_action(const wchar_t* action) override
+  {
+
+  }
+
+  virtual void enable()
+  {
+    HINSTANCE hinstance = reinterpret_cast<HINSTANCE>(&__ImageBase);
+
+    WNDCLASSEXW wcex{};
+    wcex.cbSize = sizeof(WNDCLASSEX);
+    wcex.lpfnWndProc = WndProc_Helper;
+    wcex.hInstance = hinstance;
+    wcex.lpszClassName = L"HotkeyHandleWindow";
+    RegisterClassExW(&wcex);
+
+    hotKeyHandleWindow = CreateWindowExW(WS_EX_TOOLWINDOW, L"HotkeyHandleWindow", L"", WS_POPUP, 0, 0, 0, 0, nullptr, nullptr, hinstance, this);
+    if (!hotKeyHandleWindow) {
+      return;
+    }
+
+    RegisterHotKey(hotKeyHandleWindow, 1, settings.editorHotkey.get_modifiers(), settings.editorHotkey.get_code());
+
+    enabled = true;
+  }
+
+  virtual void disable()
+  {
+    ResetCurrentOnTop();
+    enabled = false;
```

Comment:
The `hotKeyHandleWindow` should be destroyed here, and the class should be unregistered.